### DISCUSSION TITLE
Enabling SSL without using filepaths to the certificate / private key

### DISCUSSION THREAD

**Discussion by ana-m-v** [cherrypy/cherrypy#2037] [Q&A] [answered]
Hi there!

Is there any way of enabling SSL (`cherrypy.config.update`) **without** using the path to the files?
```python
'server.ssl_module':'pyopenssl',
'server.ssl_certificate':'/path_to/my_cert.crt',
'server.ssl_private_key':'/path_to/my_cert.key',
```
Could I pass the ssl context instead that might be initialized with in-memory files?

Thanks!

**Comment by webknjaz** [comment, marked as answer]
I think that the pyopessl adapter variant supports passing-in a custom context object: https://github.com/cherrypy/cheroot/blob/1ff20b1/cheroot/ssl/pyopenssl.py#L258-L261.

**Reply by ana-m-v** [reply]
Thank you! I was also considering passing the context with certificate/key directly to the update method to bypass using the paths, didn't work with built in ssl (load_cert_path...), but seems like the pyopenssl allows for in-memory files... thanks for the link, should do the trick one way or another

**Reply by webknjaz** [reply]
My understanding is that you want something like

```python
cherrypy.config.update({
    'server.ssl_module': 'pyopenssl',
    'server.ssl_context': OpenSSL.SSL.Context(...),
    'server.ssl_certificate': None,
    'server.ssl_private_key': None,
)
```

off the top of my head. I've been meaning to redesign this interface multiple times, but never got to it.
